test/unittests/commands/xmlupload/models/deserialize/test_deserialise_value.py

```diff
@@ -0,0 +1,243 @@
+
+    def test_get_value_from_knora_base_prop_hasDescription(self) -> None:
+        string = """
+            <hasDescription linkUUID="123456789">
+                text <a class="salsah-link" href="IRI:test_thing_2:IRI">link</a>
+            </hasDescription>
+        """
+        xmltext = FormattedTextValue("""text <a class="salsah-link" href="IRI:test_thing_2:IRI">link</a>""")
+        expected = XMLValue(xmltext, resrefs=["test_thing_2"], link_uuid="123456789")
+        res = XMLProperty._get_value_from_knora_base_prop(etree.fromstring(string))
+        assert res == expected
+
+    def test_get_value_from_knora_base_prop_hasComment(self) -> None:
+        string = "<hasComment>text <strong>bold</strong></hasComment>"
+        expected = XMLValue(FormattedTextValue("text <strong>bold</strong>"))
+        res = XMLProperty._get_value_from_knora_base_prop(etree.fromstring(string))
+        assert res == expected
+
+    def test_get_value_from_knora_base_prop_hasTitle(self) -> None:
+        string = "<hasTitle> Title  of   video  segment </hasTitle>"
+        expected = XMLValue("Title of video segment")
+        res = XMLProperty._get_value_from_knora_base_prop(etree.fromstring(string))
+        assert res == expected
+
+
+class Test_XMLValue:
+    def test_from_node_integer_value(self) -> None:
+        # test content, and these attributes: comments, permissions, and linkUUID (if existing)
+        node = etree.fromstring("""<integer permissions="prop-default" comment="cmt" linkUUID="foo">99</integer>""")
+        value = XMLValue.from_node(node, "integer")
+        assert value.value == "99"
+        assert value.resrefs is None
+        assert value.comment == "cmt"
+        assert value.permissions == "prop-default"
+        assert value.link_uuid == "foo"
+
+    def test_from_node_formatted_text(self) -> None:
+        # test that resrefs are extracted, and that the formatted text is tidied up
+        string = """
+            <text encoding="xml">
+                <a class="salsah-link" href="IRI:test_thing_0:IRI">first link</a>
+                This is <em>italicized and <strong>bold</strong></em> text!
+                <a class="salsah-link" href="IRI:test_thing_1:IRI">second link</a>
+            </text>
+        """
+        expected = " ".join(
+            [
+                '<a class="salsah-link" href="IRI:test_thing_0:IRI">first link</a>',
+                "This is <em>italicized and <strong>bold</strong></em> text!",
+                '<a class="salsah-link" href="IRI:test_thing_1:IRI">second link</a>',
+            ]
+        )
+        node = etree.fromstring(string)
+        value = XMLValue.from_node(node, "text")
+        assert isinstance(value.value, FormattedTextValue)
+        assert value.value.xmlstr == expected
+        assert value.resrefs == unordered(["test_thing_0", "test_thing_1"])
+        assert value.comment is None
+        assert value.permissions is None
+        assert value.link_uuid is None
```

Comment:
This is a somewhat brittle and rigid pattern for testing:
1. If you add another property to the class, you have to remember to add it to the test, otherwise it will not be tested
2. If you change the structure of the class, you manually have to change the test (which, if done too often in a code base, might demotivate people from refactoring)

Alternatively, now that it is possible to create `XMLValue` instances without using the factory method that is under test here, you could create an "expected" object and compare res against expected (like you did some lines above).  
This has the downside of not being quite as precise a test (if someone were to break the default for `comment` or `permission`, that would not be caught here - but since this is not really the aspect under test here, I think that's ok) but IMO the advantages of this outweighs the downside.

Additionally - but I'm aware that this is out of scope of this refactoring here - ideally, the factory method would have less logic, and things like resrefs and formatting tidying would be extracted into a specific function for deserializing formatted text values. Then it would be good enough to test here, that the factory correctly creates a `FormattedTextValue` and the intricacies of this creation (resrefs, formatting) could be left to tests testing this function in isolation.

Reply:
Thanks for these insightful considerations. I have changed it.